### PR TITLE
Create experiment for paywall click behavior

### PR DESCRIPTION
**Pull request by mborof**
Create experiment flag for background click behavior.

b/300159984  cl/571959672

### CODE REVIEW

File: src/runtime/experiment-flags.ts [hunk 1]
```diff
@@ -36,6 +36,13 @@ export enum ExperimentFlags {
    * Experiment flag to enable populating the client config in the Classic runtime.
    */
   POPULATE_CLIENT_CONFIG_CLASSIC = 'populate-client-config-classic',
+
+  /**
+   * Experiment flag to enable paywall background click behavior so that links
+   * cannot be clicked through the darkened background and so that it closes
+   * closable popups.
+   */
+  ENABLE_PAYWALL_BACKGROUND_CLICK = 'enable-paywall-background-click',
```
Comment (fikus) [inline src/runtime/experiment-flags.ts:45]: See note on L20 about new flags (they need to be added to E2E config).

File: src/runtime/offers-flow.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ import {
   SubscribeResponse,
   ViewSubscriptionsResponse,
 } from '../proto/api_messages';
+import {ExperimentFlags} from './experiment-flags';
```
Comment (fikus) [inline src/runtime/offers-flow.ts:26]: Changes to this file need tests.

**Review by fikus — CHANGES_REQUESTED** [review]
As I mentioned in chat, we should prefer server-controlled flags where possible. Can you update the design doc with the experiment flag rollout plan?

File: src/runtime/offers-flow.ts (hunk 2 above)
Comment (mborof) [reply src/runtime/offers-flow.ts]: I was just validating which method to use for the experiment for this control

File: src/runtime/experiment-flags.ts (hunk 1 above)
Comment (mborof) [reply src/runtime/experiment-flags.ts]: done

**Pull request merged**